src/vm/codeman.cpp

```diff
@@ -3899,19 +3907,24 @@ PTR_RUNTIME_FUNCTION EEJitManager::LazyGetFunctionEntry(EECodeInfo * pCodeInfo)
     // We need the module base address to calculate the end address of a function from the functionEntry.
     // Thus, save it off right now.
     TADDR baseAddress = pCodeInfo->GetModuleBase();
+    PTR_RUNTIME_FUNCTION pFunctionEntry = NULL;
 
     // NOTE: We could binary search here, if it would be helpful (e.g., large number of funclets)
     for (UINT iUnwindInfo = 0; iUnwindInfo < pHeader->GetNumberOfUnwindInfos(); iUnwindInfo++)
     {
-        PTR_RUNTIME_FUNCTION pFunctionEntry = pHeader->GetUnwindInfo(iUnwindInfo);
+        pFunctionEntry = pHeader->GetUnwindInfo(iUnwindInfo);
 
         if (RUNTIME_FUNCTION__BeginAddress(pFunctionEntry) <= address && address < RUNTIME_FUNCTION__EndAddress(pFunctionEntry, baseAddress))
         {
-            return pFunctionEntry;
+            break;
         }
     }
 
-    return NULL;
+#if defined(EXCEPTION_DATA_SUPPORTS_FUNCTION_FRAGMENTS)
+    pFunctionEntry = FindRootEntry(pFunctionEntry, baseAddress);
+#endif
+
+    return pFunctionEntry;
```

Comment:
Actually, I've just realized that in case we don't find a function entry, we were returning NULL before. But with your change, we would use the latest function entry we've seen. Could you please fix that to get the old behavior back? It seems that the easiest way is to move the code that looks for the root entry to the place where you have the `break`, return from there instead of breaking and put back the `return NULL` at the end.

Reply:
Thanks, will fix.